ui/src/app/components/buttons.tsx

```diff
@@ -213,7 +214,7 @@ export const Button = ({type = 'primary', style = {}, disabled = false, ...props
   />;
 };
 
-export const MenuItem = ({icon = null, tooltip = '', disabled = false, children, style = {}, ...props}) => {
+export const MenuItem = ({icon = null, faIcon = null, tooltip = '', disabled = false, children, style = {}, ...props}) => {
```

Comment:
nit: add `IconDefinition` type to `faIcon`

Reply:
I got this working. However, it's a substantial deviation from the rest of the file, so I'd rather apply the change holistically to these components in a separate PR. Here is what it would look like, let me know what you think:

```
export const MenuItem = ({icon = null, faIcon = null, tooltip = '', disabled = false, children, style = {}, ...props}: {
  icon?: string,
  faIcon?: IconDefinition,
  tooltip?: string,
  disabled?: boolean,
  children: React.ReactNode
  // Arbitrary additional props
  // This can probably be typed more strictly, based on a base React component properties type
  [key: string]: any
}) => 
```

Reply:
Oh OK. Thanks for looking into that, didn't know it would balloon into something larger.  Shipping it separately sounds good.